src/include/miopen/cat/problem_description.hpp

```diff
@@ -0,0 +1,135 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace cat {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const int32_t xCount_,
+                       const TensorDescriptor* const* xDescs_,
+                       const TensorDescriptor& yDesc_,
+                       int32_t dim_)
+        : xDescs(xDescs_), yDesc(yDesc_), xCount(xCount_), dim(dim_)
+    {
+        if((dim < 0) || (dim >= yDesc.GetLengths().size()))
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "CatForward: Dim is less than 0 or greater than max tensor dimension.");
+        }
+
+        const auto dtype = yDesc.GetType();
+        for(int i = 0; i < xCount; i++)
+        {
+            if(xDescs[i]->GetType() != dtype)
+            {
+                MIOPEN_THROW(miopenStatusBadParm, "CatForward: Tensor types do not match.");
+            }
+        }
+
+        auto ydims = yDesc.GetLengths();
+        ydims[dim] = 0;
+        for(int i = 0; i < xCount; i++)
+        {
+            auto& xdims = xDescs[i]->GetLengths();
+
+            if(ydims.size() != xdims.size())
+            {
+                MIOPEN_THROW(miopenStatusBadParm,
+                             "CatForward: Tensor dimension lengths do not match.");
+            }
+
+            for(int j = 0; j < ydims.size(); j++)
+            {
+                if((j != dim) && (ydims[j] != xdims[j]))
+                {
+                    MIOPEN_THROW(miopenStatusBadParm,
+                                 "CatForward: Tensor dimension lengths do not match.");
+                }
+            }
+            ydims[dim] += xdims[dim];
+        }
+
+        if(ydims[dim] != yDesc.GetLengths()[dim])
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "CatForward: Tensor dimension lengths do not match.");
+        }
+    }
+
+    const TensorDescriptor& GetXDesc(int i) const
+    {
+        if(i >= xCount)
+        {
+            MIOPEN_THROW(miopenStatusBadParm, "CatForward: Invalid tensor index.");
+        }
+        return *xDescs[i];
+    }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+    int32_t GetDim() const { return dim; }
+    int32_t GetXCount() const { return xCount; }
+
+    bool IsAllPacked() const
+    {
+        for(int i = 0; i < xCount; i++)
+        {
+            if(!xDescs[i]->IsPacked())
+            {
+                return false;
+            }
+        }
+
+        if(!yDesc.IsPacked())
+        {
+            return false;
+        }
+        return true;
+    }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+private:
+    const TensorDescriptor* const* xDescs;
+    const TensorDescriptor& yDesc;
```

Comment:
```suggestion
    const TensorDescriptor yDesc;
```
I overlooked it, but it is unsafe.
Check the following comments:
https://github.com/ROCm/MIOpen/pull/2671/files#r1466920140
https://github.com/ROCm/MIOpen/pull/2671/files#r1467003688

Reply:
I understand the possibility of unintended mistakes and have implemented the changes as suggested.